File: common/src/main/java/generations/gg/generations/core/generationscore/world/level/block/entities/GenerationsBlockEntityModels.java
package generations.gg.generations.core.generationscore.world.level.block.entities;

import generations.gg.generations.core.generationscore.GenerationsCore;
import generations.gg.generations.core.generationscore.world.level.block.GenerationsVoxelShapes;
import net.minecraft.core.Direction;
import net.minecraft.resources.ResourceLocation;
import net.minecraft.world.phys.shapes.Shapes;
import net.minecraft.world.phys.shapes.VoxelShape;

import static net.minecraft.world.phys.shapes.BooleanOp.OR;

public class GenerationsBlockEntityModels {
    public static final ResourceLocation DEFAULT = GenerationsCore.id("");
    public static final ResourceLocation ABUNDANT_SHRINE = block("shrines/abundant_shrine.pk");
    public static final ResourceLocation BOX = block("decorations/box/box.pk");
    public static final ResourceLocation CELESTIAL_ALTAR = block("shrines/celestial_altar.pk");
    public static final ResourceLocation COOKING_POT = block("utility_blocks/cooking_pot.pk");
    public static final ResourceLocation MOLTRES_SHRINE = block("shrines/moltres_shrine.pk");
    public static final ResourceLocation ARTICUNO_SHRINE = block("shrines/articuno_shrine.pk");
    public static final ResourceLocation GROUDON_SHRINE = block("shrines/weather_trio/groudon_shrine.pk");
    public static final ResourceLocation HOUSE_LAMP = block("decorations/house_lamp.pk");
    public static final ResourceLocation KYOGRE_SHRINE = block("shrines/weather_trio/kyogre_shrine.pk");
    public static final ResourceLocation LUGIA_SHRINE = block("shrines/lugia_shrine.pk");
    public static final ResourceLocation DARK_MODEL = block("shrines/lunar_shrine/dark_shrine.pk");
    public static final ResourceLocation LIGHT_MODEL = block("shrines/lunar_shrine/light_shrine.pk");
    public static final ResourceLocation MELOETTA_MUSIC_BOX = block("shrines/meloetta_music_box.pk");
    public static final ResourceLocation PASTEL_BEAN_BAG = block("decorations/pastel_bean_bag.pk");
    public static final ResourceLocation ROTOM_PC = block("utility_blocks/rotom_pc.pk");
    public static final ResourceLocation TABLE_PC = block("utility_blocks/table_pc.pk");
    public static final ResourceLocation POKECENTER_SCARLET_SIGN = block("decorations/pokecenter_scarlet_sign.pk");
    public static final ResourceLocation RAYQUAZA_SHRINE = block("shrines/weather_trio/groudon_shrine.pk"); //TODO: Change to Rayquaza when model is available
    public static final ResourceLocation REGICE_SHRINE = block("shrines/regis/regice_shrine.pk");
    public static final ResourceLocation REGIDRAGO_SHRINE = block("shrines/regis/regidrago_shrine.pk");
    public static final ResourceLocation REGIELEKI_SHRINE = block("shrines/regis/regieleki_shrine.pk");
    public static final ResourceLocation REGIGIGAS_SHRINE = block("shrines/regis/regigigas_shrine.pk");
    public static final ResourceLocation REGIROCK_SHRINE = block("shrines/regis/regirock_shrine.pk");
    public static final ResourceLocation REGISTEEL_SHRINE = block("shrines/regis/registeel_shrine.pk");
    public static final ResourceLocation SNORLAX_BEAN_BAG = block("decorations/snorlax_bean_bag.pk");
    public static final ResourceLocation ZAPDOS_SHRINE = block("shrines/zapdos_shrine.pk");
    public static final ResourceLocation SWITCH = block("decorations/switch.pk");
    public static final ResourceLocation TAO_TRIO_SHRINE = block("shrines/tao_trio_shrine.pk");
    public static final ResourceLocation TAPU_SHRINE = block("shrines/tapu_shrine.pk");
    public static final ResourceLocation TIME_SPACE_ALTAR = block("shrines/creation_trio/time_space_altar.pk");
    public static final ResourceLocation TRASH_CAN  = block("decorations/trash_can.pk");
    public static final ResourceLocation VENDING_MACHINE = block("utility_blocks/vending_machine.pk");
    public static final ResourceLocation BENCH = block("decorations/bench.pk");
    public static final ResourceLocation SHELF = block("decorations/shelf.pk");
    public static final ResourceLocation BOOKSHELF = block("decorations/bookshelf.pk");
    public static final ResourceLocation COUCH = block("decorations/couch.pk");
    public static final ResourceLocation DESK = block("decorations/desk.pk");
    public static final ResourceLocation STREET_LAMP = block("decorations/street_lamp.pk");
    public static final ResourceLocation DOUBLE_STREET_LAMP = block("decorations/double_street_lamp.pk");
    public static final ResourceLocation FLOOR_CUSHION = block("decorations/floor_cushion.pk");
    public static final ResourceLocation FRIDGE = block("decorations/fridge.pk");
    public static final ResourceLocation LITWICK_CANDLE = block("decorations/litwick_candle.pk");
    public static final ResourceLocation LITWICK_CANDLES = block("decorations/litwick_candles.pk");
    public static final ResourceLocation POKEBALL_PILLAR = block("decorations/pokeball_pillar.pk");
    public static final ResourceLocation BUSH = block("decorations/bush.pk");
    public static final ResourceLocation HDTV = block("decorations/hdtv.pk");
    public static final ResourceLocation SWIVEL_CHAIR = block("decorations/swivel_chair.pk");
    public static final ResourceLocation WORK_DESK = block("decorations/work_desk.pk");
    public static final ResourceLocation SHOP = block("decorations/shop.pk");
    public static final ResourceLocation BALL_DISPLAY = block("decorations/ball_display.pk");
    public static final ResourceLocation POKEBALL = block("utility_blocks/pokeball.pk");
    public static final ResourceLocation BREEDER = block("utility_blocks/breeder/breeder.pk");
    public static final ResourceLocation RKS_MACHINE = block("utility_blocks/rks_machine.pk");
    public static final ResourceLocation LIGHT_CRYSTAL = block("shrines/light_crystal.pk");
    public static final ResourceLocation DARK_CRYSTAL = block("shrines/dark_crystal.pk");
    public static final ResourceLocation SCARECROW = block("utility_blocks/scarecrow.pk");
    public static ResourceLocation HO_OH_SHRINE = block("shrines/ho_oh_shrine.pk");;

    public static ResourceLocation block(String path) {
        return GenerationsCore.id("models/block/" + path);
    }
}

File: src/common/main/kotlin/generations/gg/generations/core/generationscore/common/world/level/block/entities/GenerationsBlockEntityModels.kt
package generations.gg.generations.core.generationscore.common.world.level.block.entities

import generations.gg.generations.core.generationscore.common.GenerationsCore
import net.minecraft.resources.ResourceLocation

object GenerationsBlockEntityModels {
    val DEFAULT: ResourceLocation = GenerationsCore.id("")
    val ABUNDANT_SHRINE: ResourceLocation = block("shrines/abundant_shrine.pk")
    val BOX: ResourceLocation = block("decorations/box/box.pk")
    @JvmField
    val CELESTIAL_ALTAR: ResourceLocation = block("shrines/celestial_altar.pk")
    @JvmField
    val COOKING_POT: ResourceLocation = block("utility_blocks/cooking_pot.pk")
    val MOLTRES_SHRINE: ResourceLocation = block("shrines/moltres_shrine.pk")
    val ARTICUNO_SHRINE: ResourceLocation = block("shrines/articuno_shrine.pk")
    @JvmField
    val GROUDON_SHRINE: ResourceLocation = block("shrines/weather_trio/groudon_shrine.pk")
    val HOUSE_LAMP: ResourceLocation = block("decorations/house_lamp.pk")
    @JvmField
    val KYOGRE_SHRINE: ResourceLocation = block("shrines/weather_trio/kyogre_shrine.pk")
    @JvmField
    val LUGIA_SHRINE: ResourceLocation = block("shrines/lugia_shrine.pk")
    val DARK_MODEL: ResourceLocation = block("shrines/lunar_shrine/dark_shrine.pk")
    val LIGHT_MODEL: ResourceLocation = block("shrines/lunar_shrine/light_shrine.pk")
    @JvmField
    val MELOETTA_MUSIC_BOX: ResourceLocation = block("shrines/meloetta_music_box.pk")
    val PASTEL_BEAN_BAG: ResourceLocation = block("decorations/pastel_bean_bag.pk")
    val ROTOM_PC: ResourceLocation = block("utility_blocks/rotom_pc.pk")
    val TABLE_PC: ResourceLocation = block("utility_blocks/table_pc.pk")
    val POKECENTER_SCARLET_SIGN: ResourceLocation = block("decorations/pokecenter_scarlet_sign.pk")
    val RAYQUAZA_SHRINE: ResourceLocation = block("shrines/weather_trio/groudon_shrine.pk") //TODO: Change to Rayquaza when model is available
    @JvmField
    val REGICE_SHRINE: ResourceLocation = block("shrines/regis/regice_shrine.pk")
    @JvmField
    val REGIDRAGO_SHRINE: ResourceLocation = block("shrines/regis/regidrago_shrine.pk")
    @JvmField
    val REGIELEKI_SHRINE: ResourceLocation = block("shrines/regis/regieleki_shrine.pk")
    @JvmField
    val REGIGIGAS_SHRINE: ResourceLocation = block("shrines/regis/regigigas_shrine.pk")
    @JvmField
    val REGIROCK_SHRINE: ResourceLocation = block("shrines/regis/regirock_shrine.pk")
    @JvmField
    val REGISTEEL_SHRINE: ResourceLocation = block("shrines/regis/registeel_shrine.pk")
    val SNORLAX_BEAN_BAG: ResourceLocation = block("decorations/snorlax_bean_bag.pk")
    val ZAPDOS_SHRINE: ResourceLocation = block("shrines/zapdos_shrine.pk")
    @JvmField
    val SWITCH: ResourceLocation = block("decorations/switch.pk")
    @JvmField
    val TAO_TRIO_SHRINE: ResourceLocation = block("shrines/tao_trio_shrine.pk")
    @JvmField
    val TAPU_SHRINE: ResourceLocation = block("shrines/tapu_shrine.pk")
    @JvmField
    val TIME_SPACE_ALTAR: ResourceLocation = block("shrines/creation_trio/time_space_altar.pk")
    @JvmField
    val TRASH_CAN: ResourceLocation = block("decorations/trash_can.pk")
    @JvmField
    val VENDING_MACHINE: ResourceLocation = block("utility_blocks/vending_machine.pk")
    val BENCH: ResourceLocation = block("decorations/bench.pk")
    @JvmField
    val SHELF: ResourceLocation = block("decorations/shelf.pk")
    val BOOKSHELF: ResourceLocation = block("decorations/bookshelf.pk")
    val COUCH: ResourceLocation = block("decorations/couch.pk")
    val DESK: ResourceLocation = block("decorations/desk.pk")
    val STREET_LAMP: ResourceLocation = block("decorations/street_lamp.pk")
    @JvmField
    val DOUBLE_STREET_LAMP: ResourceLocation = block("decorations/double_street_lamp.pk")
    val FLOOR_CUSHION: ResourceLocation = block("decorations/floor_cushion.pk")
    @JvmField
    val FRIDGE: ResourceLocation = block("decorations/fridge.pk")
    val LITWICK_CANDLE: ResourceLocation = block("decorations/litwick_candle.pk")
    val LITWICK_CANDLES: ResourceLocation = block("decorations/litwick_candles.pk")
    @JvmField
    val POKEBALL_PILLAR: ResourceLocation = block("decorations/pokeball_pillar.pk")
    @JvmField
    val BUSH: ResourceLocation = block("decorations/bush.pk")
    @JvmField
    val HDTV: ResourceLocation = block("decorations/hdtv.pk")
    val SWIVEL_CHAIR: ResourceLocation = block("decorations/swivel_chair.pk")
    @JvmField
    val WORK_DESK: ResourceLocation = block("decorations/work_desk.pk")
    @JvmField
    val SHOP: ResourceLocation = block("decorations/shop.pk")
    val BALL_DISPLAY: ResourceLocation = block("decorations/ball_display.pk")
    val POKEBALL: ResourceLocation = block("utility_blocks/pokeball.pk")
    @JvmField
    val BREEDER: ResourceLocation = block("utility_blocks/breeder/breeder.pk")
    val RKS_MACHINE: ResourceLocation = block("utility_blocks/rks_machine.pk")
    @JvmField
    val LIGHT_CRYSTAL: ResourceLocation = block("shrines/light_crystal.pk")
    @JvmField
    val DARK_CRYSTAL: ResourceLocation = block("shrines/dark_crystal.pk")
    @JvmField
    val SCARECROW: ResourceLocation = block("utility_blocks/scarecrow.pk")
    @JvmField
    val PRISON_BOTTLE: ResourceLocation = block("shrines/prison_bottle.pk")
    var HO_OH_SHRINE: ResourceLocation = block("shrines/ho_oh_shrine.pk")
    var PC: ResourceLocation = block("utility_blocks/pc.pk")

    fun block(path: String): ResourceLocation {
        return GenerationsCore.id("models/block/$path")
    }
}
